### PR TITLE
refactor:description and tests for authors workshop step 01

### LINKED ISSUE #57070: Update description and tests for  step 1 of authors page workshop
We need to update a few things in step 1 of the authors page workshop

Here is the file 

https://github.com/freeCodeCamp/freeCodeCamp/blob/main/curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md

## Update description 

It always works better for the first step of these workshops to provide some context on what they will build. It is a better learning experience for campers.
Also, we shouldn't care if they use `const` or `getElementById` here. As long as they access the correct elements and assign them to the variables, that is all that matters.

Let's go with this update

```md
In this workshop, you will learn how to work with the `fetch` API to load data from an external source and display it on a page filled with freeCodeCamp authors.

All the HTML and CSS for this workshop has been provided for you. You can take a look at the two files to familiarize yourself with them.

When you are ready, start by accessing the `#author-container` and `#load-more-btn` elements. Remember you can use the `getElementById` or `querySelector` methods for this.

Then assign these elements to variables  called `authorContainer` and `loadMoreBtn`, respectively.
```

## update tests

Since we are no longer caring that campers use `const` to declare the variables or `getElementById`, all of the tests can be updated to use specific assert methods instead of regex.

https://www.chaijs.com/api/assert/

Here are the four things we should test for so it is granular:

- You should have a variable called `authorContainer`.
- You should assign the `#author-container` element to the `authorContainer` variable.
- You should have a variable called `loadMoreBtn`.
- You should assign the `#load-more-btn` element to the `loadMoreBtn` variable.

All of the tests for that don't need regex

### PR DESCRIPTION
**Pull request by RGHANILOO**
Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/how-to-open-a-pull-request/).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #57070

### CODE REVIEW

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md [hunk 1]
```diff
@@ -8,36 +8,39 @@ demoType: onLoad
 
 # --description--
 
-All the HTML and CSS for this project has been provided for you. You can take a look at the two files to familiarize yourself with them.
+In this workshop, you will learn how to work with the `fetch` API to load data from an external source and display it on a page filled with freeCodeCamp authors.
 
-Start by getting the `#author-container` and `#load-more-btn` elements with the `.getElementById()` method. Assign them to the variables `authorContainer` and `loadMoreBtn`, respectively.
+All the HTML and CSS for this workshop has been provided for you. You can take a look at the two files to familiarize yourself with them.
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md:13]: small grammar change 

```suggestion
All of the HTML and CSS for this workshop has been provided for you. You can take a look at the two files to familiarize yourself with them.
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md [hunk 2]
```diff
@@ -8,36 +8,39 @@ demoType: onLoad
...
+Then assign these elements to variables  called `authorContainer` and `loadMoreBtn`, respectively.
 
 # --hints--
 
-You should use `document.getElementById()` to get the `#author-container` element.
+You should have a variable called `authorContainer`.
 
 ```js
-assert.match(code, /document\.getElementById\(\s*('|"|`)author\-container\1\s*\)/);
+assert.isDefined(authorContainer, 'The variable authorContainer should be defined');
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md:24]: we don't need comments for the second argument

```suggestion
assert.isDefined(authorContainer);
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md [hunk 3]
```diff
@@ -8,36 +8,39 @@ demoType: onLoad
...
 ```
 
-You should assign the `#author-container` element to the variable `authorContainer`. Don't forget to use `const` to declare the variable.
+You should assign the `#author-container` element to the `authorContainer` variable.
 
 ```js
-assert.match(code, /const\s+authorContainer\s*\=\s*document\.getElementById\(\s*('|"|`)author\-container\1\s*\)/);
+assert.strictEqual(authorContainer, document.querySelector('#author-container'), 'authorContainer should be assigned the #author-container element');
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md:30]:
```suggestion
assert.strictEqual(authorContainer, document.querySelector('#author-container'));
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md [hunk 4]
```diff
@@ -8,36 +8,39 @@ demoType: onLoad
...
 ```
 
-You should use `document.getElementById()` to get the `#load-more-btn` element.
+You should have a variable called `loadMoreBtn`.
 
 ```js
-assert.match(code, /document\.getElementById\(\s*('|"|`)load\-more\-btn\1\s*\)/);
+assert.isDefined(loadMoreBtn, 'The variable loadMoreBtn should be defined');
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md:36]:
```suggestion
assert.isDefined(loadMoreBtn);
```

File: curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md [hunk 5]
```diff
@@ -8,36 +8,39 @@ demoType: onLoad
...
 ```
 
-You should assign the `#load-more-btn` element to the variable `loadMoreBtn`. Don't forget to use `const` to declare the variable.
+You should assign the `#load-more-btn` element to the `loadMoreBtn` variable.
+
 
 ```js
-assert.match(code, /const\s+loadMoreBtn\s*\=\s*document\.getElementById\(\s*('|"|`)load\-more\-btn\1\s*\)/);
+assert.strictEqual(loadMoreBtn, document.querySelector('#load-more-btn'), 'loadMoreBtn should be assigned the #load-more-btn element');
```
Comment (jdwilkin4) [inline curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md:43]:
```suggestion
assert.strictEqual(loadMoreBtn, document.querySelector('#load-more-btn'));
```

Comment (RGHANILOO) [reply curriculum/challenges/english/25-front-end-development/workshop-fcc-authors-page/641d9a19bff38d34d5a5edb8.md]: thanks for your feedback @jdwilkin4 
latest commit resolves all the suggestions

**Review by jdwilkin4 — APPROVED** [review]

**Review by Sembauke — APPROVED** [review]

**Pull request merged**